### ISSUE TITLE
4Kn drive: request not aligned to the logical block size

### ISSUE THREAD

**Issue by stintel** [koverstreet/bcachefs#58]
I have a brand new 4Kn drive formatted with bcachefs:

```
ministore /usr/src/bcachefs-tools # smartctl -i /dev/disk/by-id/wwn-0x5000cca27075ebbc
smartctl 7.0 2018-12-30 r4883 [x86_64-linux-5.1.21-00144-gba0408e8cac7] (local build)
Copyright (C) 2002-18, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Vendor:               HGST
Product:              HUH721212AL4200
Revision:             A3D0
Compliance:           SPC-4
User Capacity:        12,000,138,625,024 bytes [12.0 TB]
Logical block size:   4096 bytes
LU is fully provisioned
Rotation Rate:        7200 rpm
Form Factor:          3.5 inches
Logical Unit id:      0x5000cca27075ebbc
Serial number:        8HJ2VE2H
Device type:          disk
Transport protocol:   SAS (SPL-3)
Local Time is:        Wed Aug 21 22:57:56 2019 EEST
SMART support is:     Available - device has SMART capability.
SMART support is:     Enabled
Temperature Warning:  Enabled

ministore /usr/src/bcachefs-tools # ./bcachefs format /dev/disk/by-id/wwn-0x5000cca27075ebbc
External UUID:                  cffe2813-35f7-42c6-b11d-7d314005034c
Internal UUID:                  b458e185-35b2-4ff2-a7cc-86c9c9cae09a
Label:
Version:                        10
Created:                        Wed Aug 21 22:58:02 2019
Block_size:                     4.0K
Btree node size:                256.0K
Error action:                   remount-ro
Clean:                          0
Metadata replicas:              1
Data replicas:                  1
Metadata checksum type:         crc32c (1)
Data checksum type:             crc32c (1)
Compression type:               none (0)
Foreground write target:        none
Background write target:        none
Promote target:                 none
String hash type:               siphash (2)
32 bit inodes:                  0
GC reserve percentage:          8%
Root reserve percentage:        0%
Devices:                        1 live, 1 total
Sections:                       members
Superblock size:                816

Members (size 64):
  Device 0:
    UUID:                       959b9132-c367-4927-8b2f-4d212b2264de
    Size:                       10.9T
    Bucket size:                512.0K
    First bucket:               0
    Buckets:                    22888448
    Last mount:                 (never)
    State:                      readwrite
    Group:                      (none)
    Data allowed:               journal,btree,data
    Has data:                   (none)
    Replacement policy:         lru
    Discard:                    0
initializing new filesystem
mounted with opts: (null)
ministore /usr/src/bcachefs-tools # mount /dev/disk/by-id/wwn-0x5000cca27075ebbc /mnt/bcachefs/
```

Once I start rsyncing data to it, dmesg is flooded with these messages:
```
[29465.154998] bcachefs (sde): recovering from clean shutdown, journal seq 53
[29465.751873] bcachefs (sde): mounted with opts: (null)
[29470.820662] sd 6:0:3:0: [sde] tag#503 request not aligned to the logical block size
[29470.820664] print_req_error: 22 callbacks suppressed
[29470.820665] print_req_error: I/O error, dev sde, sector 367330304 flags 4001
[29470.820666] bch2_write_endio: 53 callbacks suppressed
[29470.820667] bcachefs (sde): IO error on sde for data write
[29470.821358] sd 6:0:3:0: [sde] tag#504 request not aligned to the logical block size
[29470.821360] print_req_error: I/O error, dev sde, sector 367331231 flags 4001
[29470.821360] bcachefs (sde): IO error on sde for data write
[29470.821362] bcachefs (sde): IO error on sde for data write
[29470.824427] sd 6:0:3:0: [sde] tag#510 request not aligned to the logical block size
[29470.824428] print_req_error: I/O error, dev sde, sector 367339520 flags 4001
[29470.824428] bcachefs (sde): IO error on sde for data write
[29470.824432] sd 6:0:3:0: [sde] tag#511 request not aligned to the logical block size
[29470.824432] print_req_error: I/O error, dev sde, sector 367339527 flags 4001
[29470.824433] bcachefs (sde): IO error on sde for data write
[29470.826479] sd 6:0:3:0: [sde] tag#451 request not aligned to the logical block size
[29470.826480] print_req_error: I/O error, dev sde, sector 367344640 flags 4001
[29470.826480] bcachefs (sde): IO error on sde for data write
[29470.827144] sd 6:0:3:0: [sde] tag#452 request not aligned to the logical block size
[29470.827145] print_req_error: I/O error, dev sde, sector 367345497 flags 4001
[29470.827145] bcachefs (sde): IO error on sde for data write
[29470.827147] bcachefs (sde): IO error on sde for data write
[29470.828883] sd 6:0:3:0: [sde] tag#455 request not aligned to the logical block size
[29470.828884] print_req_error: I/O error, dev sde, sector 367349760 flags 4001
[29470.828885] bcachefs (sde): IO error on sde for data write
[29470.828891] bcachefs (sde): IO error on sde for data write
[29470.829599] sd 6:0:3:0: [sde] tag#456 request not aligned to the logical block size
[29470.829600] print_req_error: I/O error, dev sde, sector 367351713 flags 4001
[29470.834970] sd 6:0:3:0: [sde] tag#465 request not aligned to the logical block size
[29470.834971] print_req_error: I/O error, dev sde, sector 367365120 flags 4001
[29470.835547] sd 6:0:3:0: [sde] tag#466 request not aligned to the logical block size
[29470.835548] print_req_error: I/O error, dev sde, sector 367366788 flags 4001
[29472.672203] sd 6:0:3:0: [sde] tag#496 request not aligned to the logical block size
[29472.674513] sd 6:0:3:0: [sde] tag#497 request not aligned to the logical block size
[29472.676744] sd 6:0:3:0: [sde] tag#500 request not aligned to the logical block size
[29472.677249] sd 6:0:3:0: [sde] tag#501 request not aligned to the logical block size
[29472.680847] sd 6:0:3:0: [sde] tag#505 request not aligned to the logical block size
[29472.681431] sd 6:0:3:0: [sde] tag#506 request not aligned to the logical block size
[29473.165413] sd 6:0:3:0: [sde] tag#485 request not aligned to the logical block size
[29473.165919] sd 6:0:3:0: [sde] tag#486 request not aligned to the logical block size
[29473.212875] sd 6:0:3:0: [sde] tag#480 request not aligned to the logical block size
[29473.213276] sd 6:0:3:0: [sde] tag#481 request not aligned to the logical block size
```

**Comment by stintel** [comment]
[bcachefs.dump.gz](https://github.com/koverstreet/bcachefs/files/3527093/bcachefs.dump.gz)

**Comment by stintel** [comment]
This seems to happen only when copying lots of (small) files. When I copy ubuntu-18.04.1-desktop-amd64.iso, an 1953349632 byte file, I'm not seeing any of these messages. Not even after running `sync`.

**Comment by koverstreet** [comment]
This one should be fixed now (by 293f6f93ce)

**Comment by stintel** [comment]
With eb4406623bdef4e92bb8d29dd38e09d98cf90372 this message is now flooding dmesg:
```
[ 1054.317189] bcachefs (da356edf-d43a-46ce-86d3-558ce1cae691): IO error: misaligned write
```
